mobile/src/main/java/org/openhab/habdroid/core/connection/AbstractConnection.java

```diff
@@ -97,24 +97,54 @@ public String getOpenHABUrl() {
     }
 
     @Override
-    public boolean checkReachabilityInBackground() {
+    public boolean checkReachability() {
         Log.d(TAG, "Checking reachability of " + getOpenHABUrl());
+
+        Socket s = createConnectedSocket();
+
+        if (s == null) {
+            return false;
+        }
+
+        Log.d(TAG, "Socket connected");
         try {
-            URL url = new URL(getOpenHABUrl());
-            int checkPort = url.getPort();
-            if (url.getProtocol().equals("http") && checkPort == -1)
-                checkPort = 80;
-            if (url.getProtocol().equals("https") && checkPort == -1)
-                checkPort = 443;
-            Socket s = new Socket();
-            s.connect(new InetSocketAddress(url.getHost(), checkPort), 1000);
-            Log.d(TAG, "Socket connected");
             s.close();
-            return true;
-        } catch (Exception e) {
-            Log.e(TAG, e.getMessage());
-            return false;
+        } catch (IOException ignored) {
         }
+        return true;
+    }
+
+    private Socket createConnectedSocket() {
+        for (int retries = 0; retries < 10; retries++) {
+            Socket s = new Socket();
+            InetSocketAddress socketAddress = getReachabilitySocketAddress();
```

Comment:
Should probably move this out of the loop, as it's the same every time anyway. Doing so probably also would avoid the need for a separate method for this, wouldn't it?